src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnection.cs

```diff
@@ -295,9 +299,67 @@ private void CacheConnectionStringProperties()
             if (connString != null)
             {
                 _connectRetryCount = connString.ConnectRetryCount;
+
+                // Retry Logic
+                switch (connString.RetryStrategy)
+                {
+                    case "None":
+                    case "":
+                        _retryPolicy = null;
+                        break;
+
+                    case "FixedInterval":
+                        _retryPolicy = new SqlRetryPolicy<TransientErrorDetectionStrategy>(new FixedInterval(connString.RetryCount, TimeSpan.FromSeconds(connString.RetryInterval)));
+                        break;
+
+                    case "Incremental":
+                        _retryPolicy = new SqlRetryPolicy<TransientErrorDetectionStrategy>(new Incremental(connString.RetryCount, TimeSpan.FromSeconds(connString.RetryInterval), TimeSpan.FromSeconds(connString.RetryIncrement)));
+                        break;
+
+                    case "ExponentialBackoff":
+                        _retryPolicy = new SqlRetryPolicy<TransientErrorDetectionStrategy>(new ExponentialBackoff(connString.RetryCount, TimeSpan.FromSeconds(connString.RetryMinBackoff), TimeSpan.FromSeconds(connString.RetryMaxBackoff), TimeSpan.FromSeconds(connString.RetryDeltaBackoff)));
+                        break;
+                }
+
+                List<int> errorsToAdd  = new List<int>();
+                List<int> errorsToRemove = new List<int>();
+
+                if (_retryPolicy != null)
+                {
+                    if (connString.RetriableErrors != "")
+                    {
+                        foreach (string s in connString.RetriableErrors.Split(','))
+                        {
+                            int err;
+                            if (s.IndexOf('+') != -1)
+                            {
+                                if(int.TryParse(s.Replace("+",""), out err))
```

Comment:
Parsing errors should not be ignored. The validation should be as strict as possible.